### ISSUE TITLE
MinGW warning: incompatible pointer type

### ISSUE THREAD

**Issue by kmilos** [AcademySoftwareFoundation/openexr#1444]
Building 3.1.8, this is with both GCC 13.1.0 on [MSYS2 UCRT64](https://github.com/msys2/MINGW-packages/actions/runs/5186562188/jobs/9347831213#step:11:331)/MINGW64/MINGW32 and Clang 16.0.4 on CLANG64/CLANG32:
```
C:/_/B/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_dwa_compressor.h: In function 'exr_compress_buffer':
  C:/_/B/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_dwa_compressor.h:57:21: warning: passing argument 2 of 'compress2' from incompatible pointer type [-Wincompatible-pointer-types]
     57 |                     &compbufsz,
        |                     ^~~~~~~~~~
        |                     |
        |                     size_t * {aka long long unsigned int *}
  In file included from D:/M/msys64/mingw64/include/zlib.h:34,
                   from C:/_/B/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_dwa.c:117:
  D:/M/msys64/mingw64/include/zlib.h:1244:31: note: expected 'uLongf *' {aka 'long unsigned int *'} but argument is of type 'size_t *' {aka 'long long unsigned int *'}
   1244 | ZEXTERN int ZEXPORT compress2 OF((Bytef *dest,   uLongf *destLen,
        |                               ^~
```

**Comment by claudeha** [comment]
same issue also in `openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c`:

```
/home/claude/win/posix/x86_64/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c: In function ‘apply_zip_impl’:
/home/claude/win/posix/x86_64/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c:365:21: warning: passing argument 2 of ‘compress2’ from incompatible pointer type [-Wincompatible-pointer-types]
  365 |                     &compbufsz,
      |                     ^~~~~~~~~~
      |                     |
      |                     size_t * {aka long long unsigned int *}
In file included from /home/claude/win/posix/x86_64/include/zlib.h:34,
                 from /home/claude/win/posix/x86_64/src/openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c:15:
/home/claude/win/posix/x86_64/include/zlib.h:1244:31: note: expected ‘uLongf *’ {aka ‘long unsigned int *’} but argument is of type ‘size_t *’ {aka ‘long long unsigned int *’}
 1244 | ZEXTERN int ZEXPORT compress2 OF((Bytef *dest,   uLongf *destLen,
      |                               ^~
```

**Comment by kmilos** [comment]
Yep, I guess this is due to LLP64 on Windows vs LP64 everywhere else...

**Comment by claudeha** [comment]
This patch seems to fix the warning.  Seems there would be problems (in many places, not just here) with buffers >= 4GiB in size (the uLong limit), but I don't know how likely that is to occur in real life.

```diff
--- a/openexr-3.1.8.old/src/lib/OpenEXRCore/internal_dwa_compressor.h
+++ b/openexr-3.1.8/src/lib/OpenEXRCore/internal_dwa_compressor.h
@@ -41,7 +41,7 @@ static exr_result_t exr_compress_buffer (
     size_t out_bytes_avail,
     size_t *actual_out )
 {
-    size_t         compbufsz;
+    uLong          compbufsz;
 
     if (level < 0)
     {
diff --git a/openexr-3.1.8.old/src/lib/OpenEXRCore/internal_zip.c b/openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c
index f3980ae..68cc6f3 100644
--- a/openexr-3.1.8.old/src/lib/OpenEXRCore/internal_zip.c
+++ b/openexr-3.1.8/src/lib/OpenEXRCore/internal_zip.c
@@ -347,7 +347,7 @@ static exr_result_t
 apply_zip_impl (exr_encode_pipeline_t* encode)
 {
     int            level;
-    size_t         compbufsz;
+    uLong          compbufsz;
     exr_result_t   rv;
 
     rv = exr_get_zip_compression_level (
```

**Comment by cary-ilm** [comment]
@kmilos and @claudeha, could you possibly try your build again against the ``main`` branch? The underlying code has changed and hopefully the compilation no longer fails, but I'd like to confirm.

**Comment by kmilos** [comment]
Confirming the `main` branch at commit c9b6b73e no longer issues this warning w/ GCC 13.1.0 on MSYS2 UCRT64.

The only warning left (as expected?) is the deprecated `Imf_3_2::dwaCompressionLevelAttribute()`

**Comment by cary-ilm** [comment]
Thanks for the confirmation. The fix should come in the upcoming 3.2 release in a few weeks.

**Comment by claudeha** [comment]
Confirming `main` branch no longer has this warning for me.

Host is Debian Bookworm compiling with MINGW mingw-w64 10.0.0-3 
(x86_64-w64-mingw32-g++ (GCC) 12-posix).
